experimental/QuadFormAndIsom/src/embeddings.jl

```diff
@@ -290,64 +346,66 @@ function _possible_glue_orders(qM::TorQuadModule,
 #
-# So if \phi is not equivariant, we try to find an isometry in OHN
-# to compose \phi with in order to make it equivariant (mathematically
-# this is correct: \phi was created to be any anti-isometry. The new
-# one computed is another anti-isometry, and we can take any isometry of
-# HN to make the change)
-function _can_be_made_equivariant(phi::TorQuadModuleMap,
-                                  fHM::TorQuadModuleMap,
-                                  fHN::TorQuadModuleMap)
+# If such a $\phi'$ does not exist, the default map in output is $\phi$ itself.
+function _can_be_made_equivariant(
+    phi::TorQuadModuleMap,
+    fHM::TorQuadModuleMap,
+    fHN::TorQuadModuleMap,
+  )
   OHN = orthogonal_group(domain(fHN))
   fHMinOHN = OHN(compose(inv(phi), compose(fHM, phi)); check=false)
   bool, g0 = is_conjugate_with_data(OHN, fHMinOHN, OHN(fHN; check=false))
   !bool && return false, phi
-  # Now the gluing is equivariant so the direct sum of the isometries
-  # extend to the primitive extension
   phi = compose(phi, hom(OHN(g0)))
   @hassert :ZZLatWithIsom 1 OHN(compose(inv(phi), compose(fHM, phi)); check=false) == OHN(fHN; check=false)
   return true, phi
 end
 
-# In the case of extension type (:equivariant, :plain), we look for
-# representatives of orbits of isometries on the second lattice which
-# "fit" in the gluing, i.e. for which the gluing is equivariant. These
-# isometries all live in a same coset, and we identity two isometries in this
-# coset if they are conjugate by an isometry from the classifying group.
-#
-# Let N be the associated lattice.
-# - OqfN here is the image of O(N) -> O(qN) where qN is the discriminant group.
-# - HNinqN is the embedding of the glue domain HN into qN.
-# - phig is the glue map.
-# - fHM is the representation of the isometry we aim to extend on the first
-#   lattice.
-# - discrep is the map O(N) -> O(qN).
-# - stabN is the stabilizer of HN in GN, where GN is the classifying group
-#   for N (depending on the classification type chosen by the user).
-# - if `first == true`, we return only one isometry since we only want the first
-#   extension in output of the global extension algorithm.
-function _fitting_isometries(OqfN::AutomorphismGroup{TorQuadModule},
-                             HNinqN::TorQuadModuleMap,
-                             phig::TorQuadModuleMap,
-                             fHM::TorQuadModuleMap,
-                             discrep::GAPGroupHomomorphism,
-                             stabN::AutomorphismGroup{TorQuadModule},
-                             N::ZZLat,
-                             first::Bool)
+# We are given a primitive extension $M\oplus N \subseteq L$ with $N$
+# negative definite. We assume that we have fixed an isometry $f\in O(M)$.
+# We want to determine isometries $g$ of $N$ such that $f\oplus g$ preserves
+# the lattice $L$ (call them "fitting isometries").
+# * `OqfN` here is the image of $O(N)\to O(qN)$ where `qN` is the discriminant
```

Comment:
is f involved in any way here? The notation suggests it.

Reply:
Yes, as I have written below, the input `fHM` of the function is the representation of `f` on the glue subgroup `HM` inside the discriminant group of `M`. We do not see `f` directly, by just how it acts on the gluing.

Reply:
I mean  " `OqfN` here is the image of $O(N)\to O(qN)$ " 
The image of the map $O(N)\to O(qN)$ is independent of $f$.

Reply:
Ah yes, this is just a bad variable name. I just re-use the name from the only function that calls this internal. I will change it.